Evo_Game_Function

```diff
@@ -0,0 +1,316 @@
+Evo_Game <- function(nrow = 4, ncol = 4, nFood = 5, nAgent = 2, tick = 2) {
+  
+  # Generating the data structure for the first time point plot and raw spatial information
+  
+  field <- matrix(NA, nrow, ncol)
+  indField <- matrix(1:(ncol*nrow), nrow, ncol)
+  foodField <- gen_Field_Pos(field, nFood)
+  agentField <- gen_Field_Pos(field, nAgent)
+  agentField[!is.na(agentField)] <- c(1:nAgent) #arbitrary, could also just be TRUE but this way provides identity
+  
+  
+  # Set defaults for graph
+  
+  t <- 0
+  born <- nAgent
+  died <- 0
+  nAgentTotal <- nAgent
+  
+  # First image data and plot
+  
+  imgDat <- gen_Img_Dat(foodField,agentField,field)
+  imgPlot <- gen_Img_Plot(imgDat)
+  
+  # First output structures. They contain more general and spatial information and the graph.
+  
+  metaData <- meta_Data()
+  
+  foodCoord <- as.data.frame(cbind(which(!is.na(foodField), arr.ind = TRUE), 
+                                   ind = which(!is.na(foodField))))
+  
+  agentCoord <- data.frame(ID = paste0("A", 1:nAgent), as.data.frame(which(!is.na(agentField), arr.ind = TRUE)),
+                           ind = which(!is.na(agentField)))
+  
+  Time_Series <- list()
+  Time_Series[[paste0("t",t)]] <- list(Image_Plot = imgPlot, Data_Plot = imgDat, Agent_Coordinates = agentCoord, 
+                                       Food_Coordinates = foodCoord, Meta_Data = metaData)
+  
+  #Both population and individual characteristics that depend on agent specific information.
+  #They are both adjusted due to an agents state changes during the interval.
+  
+  Population <- data.frame(ID = paste0("A", 1:nAgent), Birth_Ind = agentCoord["ind"], Birth_t = t, Death_Ind = NA, Death_t = NA, Live_Duration = NA,Parent_ID = NA, Food_Consumed = 0)
+  
+  Entire_Individual <- data.frame()
+  Individual <- data.frame(t = t, agentCoord, row.d = 0, col.d = 0, sense = 0, action = 0, energy = 0, fitness = as.factor("normal"))
+  
+  #Now the first state change phase takes place: Sensing, eating(or action), energy level change and finally fitness category evaluation.
```

Comment:
sometimes you're leaving a space between the "#" and your comment, and sometimes you don't - you might want to keep that consistent. :)